processes/process_sleepspindle_parameters/process_sleepspindle_parameters

```diff
@@ -0,0 +1,202 @@
+                                     'author of this process to further understand how the process works.<BR><BR>'
+                                     ];
+    sProcess.options.info.Type    = 'label';
+    sProcess.options.info.Value   = [];
+end
+
+
+%% ===== FORMAT COMMENT =====
+function Comment = FormatComment(sProcess) %#ok<DEFNU>
+     Comment = sProcess.Comment;
+end
+
+
+%% ===== RUN =====
+function OutputFiles = Run(sProcess, sInputs) %#ok<DEFNU>
+    % Initialize returned list of files
+    OutputFiles = {};
+    Nepochs     = length(sInputs); % Numer of inputs or epochs
+    EpochN      = linspace (1,(Nepochs),(Nepochs)); % Changes the x-axis from time (s) to epoch number n. 
+    DataMat     = in_bst(sInputs(1).FileName, [], 0); % Read the first file in the list, to obtain channel size. 
+    epochSize   = size(DataMat.F);    
+    Nchannels   = epochSize(1); % Number of ALL channels (i.e. EEG, EOG, EMG and status)
+
+    % Generate matrices of zeros (Nchannels X Nepochs) to concatenate results
+    SS_Dur = zeros (Nchannels,Nepochs); % Spindle duration
+    CountP = zeros (Nchannels,Nepochs); % Count the number of peaks
+    PeakPo = zeros (Nchannels,Nepochs); % Positive peaks
+    PeakNe = zeros (Nchannels,Nepochs); % Negative peaks 
+    TimePP = zeros (Nchannels,Nepochs); % Time of max positive peak
+    TimeNP = zeros (Nchannels,Nepochs); % Time of max negative peak
+
+    % ===== LOAD THE DATA =====
+    for iNepochs = 1 : Nepochs
+       DataMat   = in_bst(sInputs(iNepochs).FileName, [], 0);
+       epochSize = size(DataMat.F);  
+       Ntime     = epochSize(2); % Number of time samples changes depending on sleep spindle
+       Data      = DataMat.F; % Actual data containing EEG, EOG and EMG data
+       Time      = DataMat.Time(end); % Time information of the recording
+       Fs        = (Ntime-1)/Time; % Calculation of Sampling frequency
+
+    % ===== PROCESS =====
+    % This is where the actual process of data manipulation and calculation takes place.
+       DataN        = Data*-1; % The negative version of the data
+       Data(end,:)  = (randi(100,1,Ntime))*10^-6; % End row which is BDF/status channel is a constant number so replaced this with randi to remove issues during findpeaks 
+       DataN(end,:) = (randi(100,1,Ntime))*10^-6;
+       Data_Chan    = zeros(1,Ntime); % Row vector with Ntime X zeros
+       DataN_Chan   = zeros(1,Ntime); % Row vector with Ntime X zeros
+
+            for i = 1 : Nchannels
+                Data_Chan(1,:)     = Data(i,:); % Extracts individual channel info from the data matrix
+                DataN_Chan(1,:)    = DataN(i,:); % Extracts individual channel info from the negative data matrix
+                [PosP, LocP]       = findpeaks(Data_Chan,Fs); % Find all the positive peaks and their locations in seconds
+                [NegP, LocN]       = findpeaks(DataN_Chan,Fs); % Find all the negative peaks and their locations in seconds
```

Comment:
This creates a copy of the data for one channel. While it may have a small impact in memory use and performance, it is not a good practice. It could be done as:

`[PosP, LocP] = findpeaks(Data(i, :), Fs);`

Reply:
amended as per your suggestion!